CRM/Report/Form.php

```diff
@@ -2637,23 +2643,22 @@ public function buildQuery($applyLimit = TRUE) {
    * Build group by clause.
    */
   public function groupBy() {
-    $groupBys = array();
     if (!empty($this->_params['group_bys']) &&
       is_array($this->_params['group_bys'])
     ) {
       foreach ($this->_columns as $tableName => $table) {
         if (array_key_exists('group_bys', $table)) {
           foreach ($table['group_bys'] as $fieldName => $field) {
             if (!empty($this->_params['group_bys'][$fieldName])) {
-              $groupBys[] = $field['dbAlias'];
+              $this->_groupByArray[] = $field['dbAlias'];
```

Comment:
So, the goal is to address php 7.1 errors. However, I think it makes sense to try to move the code in the direction of the things that work better in the extended reports extension if we are fixing these things - specifically the extension makes the overriding of reports easier by having a function that builds the groupByArray variable so the groupBy function can be overridden with less code duplication. However, I didn't want to push too heavily in that direction in this PR since the more tidy up you do the harder it is to get it reviewed